pkg/templates/templates.go

```diff
@@ -210,7 +211,7 @@ func (t *TemplateResolver) ResolveTemplate(tmplJSON []byte, context interface{})
 	glog.V(glogDefLvl).Infof("resolved template str: %v ", resolvedTemplateStr)
 	// unmarshall before returning
 
-	resolvedTemplateBytes, err := yaml.YAMLToJSON([]byte(resolvedTemplateStr))
+	resolvedTemplateBytes, err := k8syaml.YAMLToJSON([]byte(resolvedTemplateStr))
```

Comment:
why use two different yaml pkgs?  if you are going to use yaml v3 for jsonToYAML , might as well use that for yaml to json

Reply:
@ckandag this was to reduce the overall code we need to maintain since `YAMLToJSON` in the Kubernetes yaml library is not super trivial.

Reply:
I thought we were previously using  k8s yaml and doing the marshalling in fomYAML?,  is it really that complex ?

I am very Vary of using two different versions/pkg  of yaml for marshalling and unmarshalling ,  either pkg  is fine as long as you use the same for both there could be differences in how the yaml content is interpreted in the two versions  like for e.g datatypes or yaml tags etc .  I am going to let Gus provide an opinion on this.

Reply:
@ckandag Right, so the Kubernetes YAML library (i.e. `sigs.k8s.io/yaml`) uses `gopkg.in/yaml.v2` under the hood. `gopkg.in/yaml.v2` has line wrapping enabled by default. It can be disabled globally on later versions of v2, but I didn't want to alter the behavior globally. This is why I chose to use `gopkg.in/yaml.v3` for converting the JSON returned from the Kubernetes API to YAML. This is only an implementation detail that the JSON is converted to YAML since the template functions (e.g. `toInt`) rely on it being YAML and not JSON. Eventually, it'd probably be good to just support the JSON directly.

I'm not concerned with using a newer version of the same library for temporarily converting JSON to YAML because YAML is a superset of JSON. I did want to keep using the `sigs.k8s.io/yaml` library since it does some compatibility conversion for YAML to JSON, but now that I think of it, since the original form is JSON, there is no compativility conversions require, so I can just use `gopkg.in/yaml.v3` directly.

Reply:
@ckandag okay, I made the change so now only `gopkg.in/yaml.v3` is used